app/controllers/application_controller.rb

```diff
@@ -56,16 +56,22 @@ def restrict_iframes
 
   def enable_embedded_shopfront
     whitelist = Spree::Config[:embedded_shopfronts_whitelist]
-    return unless Spree::Config[:enable_embedded_shopfronts] && whitelist.present?
+    domain = embedded_shopfront_referer
+    return unless Spree::Config[:enable_embedded_shopfronts] && whitelist.present? && domain.present? && whitelist.include?(domain)
```

Comment:
IMO it's now worth extracting this now long boolean logic into a more abstract method such as `embeddable?` or something alike. It will make things easier to understand to the next dev.

Reply:
I agree, moving it to a method would also increase "documentation surface" (i.e. method documentation, tests, etc)

Reply:
Done. It looks much nicer now.